crates/sui-graphql-rpc/src/extensions/limits_info.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::{
+    extensions::{Extension, ExtensionContext, ExtensionFactory, NextRequest, NextValidation},
+    value, Response, ServerError, ValidationResult,
+};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+use crate::config::ServiceConfig;
+
+pub struct LimitsInfo;
+
+impl ExtensionFactory for LimitsInfo {
+    fn create(&self) -> Arc<dyn Extension> {
+        Arc::new(LimitsInfoExtension::default())
+    }
+}
+
+#[derive(Default)]
+struct LimitsInfoExtension {
+    validation_result: Mutex<Option<ValidationResult>>,
```

Comment:
...this is going to bottleneck all requests to the GraphQL server.  I don't think that's the right way to go.